frontend/server/src/DAO/Courses.php

```diff
@@ -91,48 +92,139 @@ public static function getAllAssignments(
     }
 
     /**
-     * @return list<\OmegaUp\DAO\VO\Courses>
+     * @return list<FilteredCourse>
      */
     public static function getCoursesForStudent(int $identityId) {
-        $sql = 'SELECT c.*
+        $sql = 'SELECT
+                    admission_mode,
+                    alias,
+                    course_id,
+                    finish_time,
+                    c.name AS name,
+                    s.name AS school_name,
+                    start_time,
+                    accept_teacher
                 FROM Courses c
                 INNER JOIN (
-                    SELECT g.group_id
+                    SELECT g.group_id, gi.accept_teacher
                     FROM Groups_Identities gi
                     INNER JOIN `Groups_` AS g ON g.group_id = gi.group_id
                     WHERE gi.identity_id = ?
                 ) gg
-                ON c.group_id = gg.group_id;
-               ';
-        /** @var list<array{acl_id: int, admission_mode: string, alias: string, course_id: int, description: string, finish_time: \OmegaUp\Timestamp|null, group_id: int, name: string, needs_basic_information: bool, requests_user_information: string, school_id: int|null, show_scoreboard: bool, start_time: \OmegaUp\Timestamp}> */
+                ON c.group_id = gg.group_id
+                LEFT JOIN
+                    Schools s
+                ON c.school_id = s.school_id;';
+        /** @var list<array{accept_teacher: bool|null, admission_mode: string, alias: string, course_id: int, finish_time: \OmegaUp\Timestamp|null, name: string, school_name: null|string, start_time: \OmegaUp\Timestamp}> */
         $rs = \OmegaUp\MySQLConnection::getInstance()->GetAll(
             $sql,
             [$identityId]
         );
+
         $courses = [];
         foreach ($rs as $row) {
-            $courses[] = new \OmegaUp\DAO\VO\Courses($row);
+            $assignmentsProgress = \OmegaUp\DAO\Courses::getAssignmentsProgress(
+                $row['course_id'],
+                $identityId
+            );
+
+            /** @var float */
+            $points = array_reduce(
+                $assignmentsProgress,
+                /**
+                 * @param float|null $accumulator
+                 * @param array{score: float, max_score: float} $item
+                 */
+                function (
+                    $accumulator,
```

Comment:
```suggestion
                    float $accumulator,
```

si le pasas un tercer argumento `0` a [`array_reduce`](http://php.net/array_reduce), podemos evitar la nulabilidad del acumulador, y además se puede reemplazar L142-L47 por

```php
  return $accumulator + $item['score'];
```

igual en todos los demás lugares donde se usa `array_reduce`. y _posiblemente_ con eso se puede eliminar la anotación de L131.

Reply:
Listo.